src/test/java/io/javalin/TestCookieStore.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Test;
+
+import io.javalin.translator.template.TemplateUtil;
+import io.javalin.util.TestObject_Serializable;
+
+import static org.hamcrest.MatcherAssert.*;
+import static org.hamcrest.Matchers.*;
+
+public class TestCookieStore extends _UnirestBaseTest {
+
+    @After
+    public void after() throws Exception {
+        clearCookies();
+    }
+
+    @Test
+    public void test_cookieStore_betweenHandlers() throws Exception {
+        app.get("/cookie-store", ctx -> ctx.cookieStore("test-object", new TestObject_Serializable()));
+        app.after("/cookie-store", ctx -> {
+            if (ctx.cookieStore("test-object") instanceof TestObject_Serializable) {
+                ctx.result("Got stored value from different handler");
+            }
+        });
+        assertThat(GET_body("/cookie-store"), is("Got stored value from different handler"));
+    }
+
+    @Test
+    public void test_cookieStore_betweenRequests() throws Exception {
+        app.get("/cookie-storer", ctx -> ctx.cookieStore("test-object", new TestObject_Serializable()));
+        app.get("/cookie-reader", ctx -> {
+            if (ctx.cookieStore("test-object") instanceof TestObject_Serializable) {
+                ctx.result("Got stored value from different request");
+            }
+        });
+        GET_body("/cookie-storer");
```

Comment:
what is actually happening here? Can't find GET_body anywhere

Reply:
`GET_body(String url);` is part of `_UnirestBaseTest`. It calls a url and returns the body of the request. In this case I'm ignoring the result, I'm just calling the endpoint in order to store the cookie value.